src/midi/midi_alsa.h

```diff
@@ -28,14 +28,17 @@
 
 #include <alsa/asoundlib.h>
 
+struct alsa_address {
+	int client;
+	int port;
+};
+
 class MidiHandler_alsa : public MidiHandler {
 private:
 	snd_seq_event_t ev = {};
 	snd_seq_t *seq_handle = nullptr;
-	int seq_client = 0;
-	int seq_port = 0;
-	int my_client = 0;
-	int my_port = 0;
+	alsa_address seq = {0, 0};  // address of input port we're connected to
```

Comment:
We use `{ -1, -1 }` in a couple places above as the invalid/unset case.
I'm not sure if it's possible to use the same defaults across the board, but just mentioning it for consistency.

Reply:
I decided to use -1, -1 in new code as  0, 0 is *formally* a valid ALSA sequencer port… I think I'll change the default to -1,-1 across the board.